crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -227,6 +227,46 @@ impl Format<PyFormatContext<'_>> for FormatDanglingComments<'_> {
     }
 }
 
+/// Formats the dangling comments within a parenthetical, for example:
+/// ```python
+/// [  # comment
+///     1,
+///     2,
+///     3,
+/// ]
+/// ```
+pub(crate) fn parenthetical_comments(comments: &[SourceComment]) -> FormatParentheticalComments {
+    FormatParentheticalComments { comments }
+}
+
+pub(crate) struct FormatParentheticalComments<'a> {
+    comments: &'a [SourceComment],
+}
+
+impl Format<PyFormatContext<'_>> for FormatParentheticalComments<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext>) -> FormatResult<()> {
+        let mut first = true;
+
+        for comment in self
+            .comments
+            .iter()
+            .filter(|comment| comment.is_unformatted())
+        {
+            if first {
+                write!(f, [space(), space()])?;
+            }
+
+            write!(f, [format_comment(comment)])?;
+
+            comment.mark_formatted();
+
+            first = false;
+        }
+
+        Ok(())
+    }
+}
```

Comment:
A dedicated formatter for these kinds of inner-parenthetical methods, rather than using the generic `dangling_comments` formatter which always hard-breaks after comments.